interfaces/udev/spec.go

```diff
@@ -163,3 +164,24 @@ func (spec *Specification) AddPermanentSlot(iface interfaces.Interface, slot *sn
 	}
 	return nil
 }
+
+// Informs ReloadRules() to also do 'udevadm trigger <subsystem specific>'.
+// IMPORTANT: because there is currently no way to call TriggerSubsystem during
+// interface disconnect, TriggerSubsystem() should typically only by used in
+// UDevPermanentSlot since the rules are permanent until the snap is removed.
+func (spec *Specification) TriggerSubsystem(subsystem string) {
+	if subsystem == "" {
+		return
+	}
+
+	for _, elem := range spec.udevadmSubsystemTriggers {
+		if subsystem == elem {
+			return
+		}
+	}
+	spec.udevadmSubsystemTriggers = append(spec.udevadmSubsystemTriggers, subsystem)
+}
+
+func (spec *Specification) GetTriggeredSubsystems() []string {
+	return spec.udevadmSubsystemTriggers
```

Comment:
This is safe currently but to avoid mutating the internal state we could consider returning a copy instead.